<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1535</width>
    <height>930</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>openPOWERViz</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>:/icons/resources/logo2_NOTTF.svg</normaloff>:/icons/resources/logo2_NOTTF.svg</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tabNet">
       <attribute name="title">
        <string>Network</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_13">
        <item row="1" column="1">
         <widget class="QScrollArea" name="networkGraph">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Wird alles dynamisch angepasst. Mit Qt und Mathematik&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Plain</enum>
          </property>
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOn</enum>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOn</enum>
          </property>
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="NetworkGraphWidget" name="networkGraphContents">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>937</width>
             <height>408</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="gridLayout_15"/>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWTPlotWidget" name="tabGraph">
       <attribute name="title">
        <string>Plot</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_14">
        <item row="0" column="0">
         <widget class="QwtPlot" name="qwtPlot"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1535</width>
     <height>30</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Fi&amp;le</string>
    </property>
    <addaction name="actionLoad"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
    <addaction name="separator"/>
    <addaction name="actionStart_Recording"/>
    <addaction name="actionStop_Recording"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionSettings"/>
    <addaction name="actionPlugins"/>
    <addaction name="actionInterfaces"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <widget class="QMenu" name="menuWidgets">
     <property name="title">
      <string>W&amp;idgets</string>
     </property>
     <addaction name="actionStuff1"/>
     <addaction name="actionStuff2"/>
    </widget>
    <widget class="QMenu" name="menuOD_Filter">
     <property name="title">
      <string>&amp;OD Filter</string>
     </property>
     <addaction name="actionALL"/>
     <addaction name="actionFancy"/>
     <addaction name="actionBest"/>
    </widget>
    <addaction name="actionFullscreen"/>
    <addaction name="separator"/>
    <addaction name="menuWidgets"/>
    <addaction name="menuOD_Filter"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionLicence"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionLoad"/>
   <addaction name="actionSave"/>
   <addaction name="actionSave_As"/>
   <addaction name="separator"/>
   <addaction name="actionStart_Recording"/>
   <addaction name="actionStop_Recording"/>
   <addaction name="separator"/>
   <addaction name="actionOD_Filter_2"/>
  </widget>
  <widget class="TimelineWidget" name="dockTime">
   <property name="baseSize">
    <size>
     <width>0</width>
     <height>100</height>
    </size>
   </property>
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="allowedAreas">
    <set>Qt::AllDockWidgetAreas</set>
   </property>
   <property name="windowTitle">
    <string>Ti&amp;meline</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_6">
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="0">
      <widget class="QwtPlot" name="qwtPlotTimeline">
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>100</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="TimelineWidget" name="dockPlugins">
   <property name="windowTitle">
    <string>Pl&amp;ugins</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QGridLayout" name="gridLayout_5">
     <item row="0" column="0">
      <widget class="QPushButton" name="pushButton">
       <property name="text">
        <string>Plugin Editor</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="PluginSelectorWidget" name="pluginSelectorWidget">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Plugins koennen hierr ein und ausgeschaltet werden.&lt;/p&gt;&lt;p&gt;Auch das overlay in der Timeline fuer den Plugin kann hier getoggelt werden.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="CurrentODWidget" name="dockCurrent">
   <property name="windowTitle">
    <string>Current &amp;Node OD</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QTreeView" name="curNodeODWidget"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="ODDescriptionWidget" name="dockOD">
   <property name="windowTitle">
    <string>OD Desc&amp;riptions</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QGridLayout" name="gridLayout_4">
     <item row="0" column="0">
      <widget class="QTreeWidget" name="odDescriptionWidget">
       <column>
        <property name="text">
         <string>Index</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Name</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Type</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Description</string>
        </property>
       </column>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockEvents">
   <property name="windowTitle">
    <string>Events</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockEventsContents">
    <layout class="QGridLayout" name="gridLayout_6">
     <item row="0" column="0">
      <widget class="EventViewerWidget" name="eventLog">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Unterstuetzt das automatische Springen zu cyclen&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <column>
        <property name="text">
         <string>Event</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Description</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Type</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Flags</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>First</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Last</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Plugin</string>
        </property>
       </column>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="CycleCommandsWidget" name="dockPaketView">
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Zeigt die commands des ausgwaehlten Cycles an&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="windowTitle">
    <string>C&amp;ycle Commands</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_7">
    <layout class="QGridLayout" name="gridLayout_18">
     <item row="0" column="0">
      <widget class="QTreeView" name="cycleCommandsView"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="PythonLogWidget" name="dockPythonLog">
   <property name="windowTitle">
    <string>Python Lo&amp;g</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_9">
    <layout class="QGridLayout" name="gridLayout_20">
     <item row="0" column="0">
      <widget class="QTableView" name="pythonLogView"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="PacketHistoryWidget" name="dockPacketHistory">
   <property name="windowTitle">
    <string>Pac&amp;ketHistory</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_10">
    <layout class="QGridLayout" name="gridLayout_21">
     <item row="0" column="0">
      <widget class="QScrollArea" name="scrollArea_4">
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents_4">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>749</width>
          <height>96</height>
         </rect>
        </property>
        <layout class="QGridLayout" name="gridLayout_22">
         <item row="0" column="0">
          <widget class="QPlainTextEdit" name="packetHistoryTextEdit">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="textInteractionFlags">
            <set>Qt::NoTextInteraction</set>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionQuit">
   <property name="icon">
    <iconset>
     <normalon>:/icons/resources/application-exit.svg</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
   <property name="menuRole">
    <enum>QAction::QuitRole</enum>
   </property>
  </action>
  <action name="actionLoad">
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/resources/document-open-symbolic.symbolic.png</normaloff>:/icons/resources/document-open-symbolic.symbolic.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Load</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/resources/document-save-symbolic.symbolic.png</normaloff>:/icons/resources/document-save-symbolic.symbolic.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/resources/document-save-as-symbolic.symbolic.png</normaloff>:/icons/resources/document-save-as-symbolic.symbolic.png</iconset>
   </property>
   <property name="text">
    <string>Sa&amp;ve As</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset>
     <normalon>:/icons/resources/help-about.svg</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>&amp;About</string>
   </property>
   <property name="menuRole">
    <enum>QAction::AboutRole</enum>
   </property>
  </action>
  <action name="actionLicence">
   <property name="text">
    <string>&amp;Licence</string>
   </property>
  </action>
  <action name="actionFullscreen">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Fullscreen</string>
   </property>
   <property name="shortcut">
    <string>F11</string>
   </property>
  </action>
  <action name="actionStuff1">
   <property name="text">
    <string>&amp;Stuff1</string>
   </property>
  </action>
  <action name="actionStuff2">
   <property name="text">
    <string>St&amp;uff2</string>
   </property>
  </action>
  <action name="actionStart_Recording">
   <property name="text">
    <string>Start R&amp;ecording</string>
   </property>
  </action>
  <action name="actionStop_Recording">
   <property name="text">
    <string>Stop &amp;Recording</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="text">
    <string>&amp;Settings</string>
   </property>
  </action>
  <action name="actionPlugins">
   <property name="text">
    <string>&amp;Plugins</string>
   </property>
  </action>
  <action name="actionALL">
   <property name="text">
    <string>&amp;ALL</string>
   </property>
  </action>
  <action name="actionOD_Filter_2">
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/resources/view-filter.svg</normaloff>:/icons/resources/view-filter.svg</iconset>
   </property>
   <property name="text">
    <string>OD Filter</string>
   </property>
  </action>
  <action name="actionFancy">
   <property name="text">
    <string>&amp;Fancy</string>
   </property>
  </action>
  <action name="actionBest">
   <property name="text">
    <string>&amp;Best</string>
   </property>
  </action>
  <action name="actionInterfaces">
   <property name="text">
    <string>&amp;Interfaces</string>
   </property>
   <property name="toolTip">
    <string>Setup Interfaces</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QwtPlot</class>
   <extends>QFrame</extends>
   <header>qwt_plot.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>PythonLogWidget</class>
   <extends>QDockWidget</extends>
   <header>PythonLogWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TimelineWidget</class>
   <extends>QDockWidget</extends>
   <header>TimeLineWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>CurrentODWidget</class>
   <extends>QDockWidget</extends>
   <header>CurrentODWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ODDescriptionWidget</class>
   <extends>QDockWidget</extends>
   <header>ODDescriptionWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>CycleCommandsWidget</class>
   <extends>QDockWidget</extends>
   <header>CycleCommandsWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>NetworkGraphWidget</class>
   <extends>QWidget</extends>
   <header>NetworkGraphWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>PacketHistoryWidget</class>
   <extends>QDockWidget</extends>
   <header>PacketHistoryWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>PluginSelectorWidget</class>
   <extends>QListWidget</extends>
   <header>PluginSelectorWidget.hpp</header>
   <slots>
    <slot>loadPlugins(EPL_DataCollect::CaptureInstance)</slot>
    <slot>disableWidget()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>QWTPlotWidget</class>
   <extends>QWidget</extends>
   <header>QWTPlotWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>EventViewerWidget</class>
   <extends>QTreeWidget</extends>
   <header>EventViewerWidget.hpp</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>593</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionFullscreen</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>setFullscreen(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>593</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPlugins</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openPluginEditor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>604</x>
     <y>380</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>openPluginEditor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>135</x>
     <y>219</y>
    </hint>
    <hint type="destinationlabel">
     <x>604</x>
     <y>380</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setFocus()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>604</x>
     <y>380</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionInterfaces</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openInterfacePicker()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>767</x>
     <y>464</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStart_Recording</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>startRecording()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>767</x>
     <y>464</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStop_Recording</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>stopRecording()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>767</x>
     <y>464</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLoad</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>open()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>767</x>
     <y>464</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStop_Recording</sender>
   <signal>triggered()</signal>
   <receiver>pluginSelectorWidget</receiver>
   <slot>disableWidget()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>138</x>
     <y>268</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSettings</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>767</x>
     <y>464</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>openPluginEditor()</slot>
  <slot>openInterfacePicker()</slot>
  <slot>setFullscreen(bool)</slot>
  <slot>startRecording()</slot>
  <slot>stopRecording()</slot>
  <slot>open()</slot>
  <slot>openSettings()</slot>
 </slots>
</ui>
